src/vip-framework/index.ts

```diff
@@ -7,26 +7,38 @@ import { FORKED_NETWORK, ethers } from "hardhat";
 
 import { NETWORK_ADDRESSES } from "../networkAddresses";
 import { NETWORK_CONFIG } from "../networkConfig";
-import { Proposal, SUPPORTED_NETWORKS } from "../types";
-import { getCalldatas, initMainnetUser, setForkBlock } from "../utils";
+import { Proposal, ProposalType, REMOTE_NETWORKS, SUPPORTED_NETWORKS } from "../types";
+import {
+  calculateGasForAdapterParam,
+  getCalldatas,
+  getOmnichainProposalSenderAddress,
+  getPayload,
+  getSourceChainId,
+  initMainnetUser,
+  setForkBlock,
+} from "../utils";
+import ENDPOINT_ABI from "./abi/LzEndpoint.json";
+import OMNICHAIN_EXECUTOR_ABI from "./abi/OmnichainGovernanceExecutor.json";
 import GOVERNOR_BRAVO_DELEGATE_ABI from "./abi/governorBravoDelegateAbi.json";
 
 const DEFAULT_SUPPORTER_ADDRESS = "0xc444949e0054a23c44fc45789738bdf64aed2391";
+const OMNICHAIN_PROPOSAL_SENDER = getOmnichainProposalSenderAddress();
+const OMNICHAIN_GOVERNANCE_EXECUTOR =
+  NETWORK_ADDRESSES[FORKED_NETWORK as REMOTE_NETWORKS].OMNICHAIN_GOVERNANCE_EXECUTOR;
 
 const VOTING_PERIOD = 28800;
 
 export const { DEFAULT_PROPOSER_ADDRESS, GOVERNOR_PROXY, NORMAL_TIMELOCK } =
-  NETWORK_ADDRESSES[FORKED_NETWORK as "bsctestnet" | "bscmainnet"] || {};
-
+  NETWORK_ADDRESSES[(FORKED_NETWORK as "bscmainnet") || "bsctestnet"] || {};
 export const { DELAY_BLOCKS } = NETWORK_CONFIG[FORKED_NETWORK as SUPPORTED_NETWORKS];
 
-export const forking = (blockNumber: number, fn: () => void) => {
-  describe(`At block #${blockNumber}`, () => {
-    before(async () => {
-      await setForkBlock(blockNumber);
-    });
-    fn();
-  });
+export const forking = (blockNumber: number, fn: () => Promise<void>) => {
+  (async () => {
+    console.log(`At block #${blockNumber}`);
+    await setForkBlock(blockNumber);
+    await fn();
+    run();
+  })();
 };
```

Comment:
Wow, I didn't know you could do that. It looks like magic! Is it [delayed root suite](https://mochajs.org/#delayed-root-suite)?

Reply:
yes